csrc/scheduler/matmul.cpp

```diff
@@ -901,10 +901,17 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
   addSetForCacheRead(acw_smem, &acr);
   addSetForCacheRead(bcw_smem, &bcr);
 
+  const std::vector<ValGroup> ordering = mma_utils::canonicalDimOrdering(
+      tensor_roles, id_roles, id_model.idGraph(IdMappingMode::PERMISSIVE));
+
   // Make a CTA tile
   // ------------------------------------------------------------------
   // Dimensions ordered as: [ (device dims), (batch dims), M, N, K ]
-  mma_utils::canonicalizeMmaTvOrdering(mma_result);
+  mma_utils::canonicalizeMmaTvOrdering(
+      mma_result,
+      id_model.idGraph(IdMappingMode::PERMISSIVE),
+      id_roles,
+      ordering);
```

Comment:
We could probably collect the permissive graph, dim roles, tensor roles, and canonical ordering into a single struct, but I worry a little about lifetime since often when we modify the fusion definition we need to rebuild the permissive map which will invalidate the previously saved `ValGroup`s. For now, I'm just trusting the calling scope to manage all that.